kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/EventHandlerServlet.java

```diff
@@ -80,18 +74,87 @@ public void handleEvent(final Event event) {
     @Override
     public void doGet(final HttpServletRequest request, final HttpServletResponse response)
             throws ServletException, IOException {
-
         response.setContentType("text/event-stream");
         response.setCharacterEncoding("UTF-8");
         response.setHeader("Cache-control", "no-cache");
         response.setHeader("Connection", "keep-alive");
+        response.setHeader("Access-Control-Allow-Origin", "*"); // required for IE9
 
-        final PrintWriter writer = response.getWriter();
-        while (!this.m_sharedCache.isEmpty()) {
-            writer.write("data: " + this.m_sharedCache.poll() + "\n\n");
+        final Thread thread = Thread.currentThread();
+        final BlockingQueue<String> eventQueue = new LinkedBlockingQueue<>(MAX_SIZE_OF_QUEUE);
+        final OutputStream outputStream = response.getOutputStream();
+        final PrintStream printStream = new PrintStream(outputStream);
+        final AtomicReference<Closeable> closeableReference = new AtomicReference<>(outputStream);
+        final ServiceRegistration<?> registration = register(EMIT_EVENT_TOPIC, eventQueue, closeableReference, thread);
+
+        try {
+            while (true) {
+                final String data = eventQueue.poll(2, TimeUnit.SECONDS);
```

Comment:
The idea was to provide a servlet that allows javascript clients to see the Event Admin events which blocks the request thread until it gets killed. Asynchronous task processing will be beneficial in this scenario. As far as I am aware of `Asynchronous Servlet Processing` does not provide any option to cancel the ongoing task. I have to check it once. I am also thinking of using `CompletableFuture` which provides asynchronous processing with more advanced functionalities. I will make the necessary amendments and will let you know.